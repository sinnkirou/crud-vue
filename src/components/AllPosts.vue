<template>
  <div id="AllPosts">
    <div class="navbar">
      <h1 class="post_heading">All Posts</h1>
    </div>
    <ViewablePost v-for="post in posts" v-bind:key="post.id" v-bind:post="post"/>
  </div>
</template>

<script>
import ViewablePost from "./ViewablePost.vue";
export default {
  name: "AllPosts",
  props: {
    posts: Array
  },
  components: {
    ViewablePost
  }
};
</script>
