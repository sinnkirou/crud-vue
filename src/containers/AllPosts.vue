<template>
  <div>
    <div class="allPosts">
      <h1 class="post_heading">All Posts {{postsCount(keyword)}}</h1>
      <div class="mdl-textfield mdl-js-textfield">
        <label class="mdl-button mdl-js-button mdl-button--icon">
          <i class="material-icons">search</i>
        </label>
        <input
          class="mdl-textfield__input"
          type="text"
          placeholder="Enter post title to search"
          v-model="keyword"
        >
      </div>
    </div>
    <PostItem v-for="post in posts" :key="post.id" :post="post" v-show="shouldShow(post.title)"/>
  </div>
</template>

<script>
import { createNamespacedHelpers } from "vuex";
const { mapGetters, mapState } = createNamespacedHelpers("modulePosts");

export default {
  name: "AllPosts",
  computed: {
    ...mapState(["posts"]),
    ...mapGetters(["postsCount"])
  },
  data: () => ({
    keyword: ""
  }),
  components: {
    PostItem: () => import("../components/PostItem")
  },
  methods: {
    shouldShow: function(title) {
      var regex = new RegExp(this.keyword, "i");
      return !this.keyword || regex.test(title);
    }
  }
};
</script>
