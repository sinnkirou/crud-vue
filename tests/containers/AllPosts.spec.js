import { shallowMount } from "@vue/test-utils";
import AllPosts from "@/containers/AllPosts.vue";
import { localVue, store, posts } from "../mockStore";

describe("AllPosts.vue", () => {
  let wrapper;
  let vm;

  beforeEach(() => {
    wrapper = shallowMount(AllPosts, { localVue, store });
    vm = wrapper.vm;
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it("should render", () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
    expect(vm.posts).toEqual(posts);
  });

  it("has the expected html structure", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("when search successfully", () => {
    expect(vm.keyword).toEqual("");
    var input = wrapper.find("input");
    input.element.value = posts[0].title;
    input.trigger("input");
    expect(vm.keyword).toEqual(posts[0].title);
    expect(wrapper.element).toMatchSnapshot();
  });

  it("when search unsuccessfully", () => {
    expect(vm.keyword).toEqual("");
    var input = wrapper.find("input");
    var keyword = "testdsdd";
    input.element.value = keyword;
    input.trigger("input");
    expect(vm.keyword).toEqual(keyword);
    expect(wrapper.element).toMatchSnapshot();
  });
});
