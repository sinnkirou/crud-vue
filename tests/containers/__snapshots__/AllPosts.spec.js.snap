// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllPosts.vue has the expected html structure 1`] = `
<div>
  <div
    class="allPosts"
  >
    <h1
      class="post_heading"
    >
      All Posts 1
    </h1>
     
    <div
      class="mdl-textfield mdl-js-textfield"
    >
      <label
        class="mdl-button mdl-js-button mdl-button--icon"
      >
        <i
          class="material-icons"
        >
          search
        </i>
      </label>
       
      <input
        class="mdl-textfield__input"
        placeholder="Enter post title to search"
        type="text"
      />
    </div>
  </div>
   
  <div
    class="post"
  >
    <div>
      <h2
        class="post_title"
      >
        title
      </h2>
       
      <p
        class="post_message"
      >
        message
      </p>
       
      <p
        class="post_id"
      >
        id: 1
      </p>
       
      <div
        class="control_buttons"
      >
        <button
          class="edit"
          type="button"
        >
          Edit
        </button>
         
        <button
          class="delete"
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`AllPosts.vue when search successfully 1`] = `
<div>
  <div
    class="allPosts"
  >
    <h1
      class="post_heading"
    >
      All Posts 1
    </h1>
     
    <div
      class="mdl-textfield mdl-js-textfield"
    >
      <label
        class="mdl-button mdl-js-button mdl-button--icon"
      >
        <i
          class="material-icons"
        >
          search
        </i>
      </label>
       
      <input
        class="mdl-textfield__input"
        placeholder="Enter post title to search"
        type="text"
      />
    </div>
  </div>
   
  <div
    class="post"
  >
    <div>
      <h2
        class="post_title"
      >
        title
      </h2>
       
      <p
        class="post_message"
      >
        message
      </p>
       
      <p
        class="post_id"
      >
        id: 1
      </p>
       
      <div
        class="control_buttons"
      >
        <button
          class="edit"
          type="button"
        >
          Edit
        </button>
         
        <button
          class="delete"
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`AllPosts.vue when search unsuccessfully 1`] = `
<div>
  <div
    class="allPosts"
  >
    <h1
      class="post_heading"
    >
      All Posts 0
    </h1>
     
    <div
      class="mdl-textfield mdl-js-textfield"
    >
      <label
        class="mdl-button mdl-js-button mdl-button--icon"
      >
        <i
          class="material-icons"
        >
          search
        </i>
      </label>
       
      <input
        class="mdl-textfield__input"
        placeholder="Enter post title to search"
        type="text"
      />
    </div>
  </div>
   
  <div
    class="post"
    style="display: none;"
  >
    <div>
      <h2
        class="post_title"
      >
        title
      </h2>
       
      <p
        class="post_message"
      >
        message
      </p>
       
      <p
        class="post_id"
      >
        id: 1
      </p>
       
      <div
        class="control_buttons"
      >
        <button
          class="edit"
          type="button"
        >
          Edit
        </button>
         
        <button
          class="delete"
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</div>
`;
